packages/kbn-reporting/public/share/share_context_menu/register_csv_modal_reporting.tsx

```diff
@@ -0,0 +1,202 @@
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { i18n } from '@kbn/i18n';
+import React from 'react';
+import { toMountPoint } from '@kbn/kibana-react-plugin/public';
+
+import { CSV_JOB_TYPE, CSV_JOB_TYPE_V2 } from '@kbn/reporting-export-types-csv-common';
+
+import type { SearchSourceFields } from '@kbn/data-plugin/common';
+import { ShareContext, ShareMenuItem, ShareMenuProvider } from '@kbn/share-plugin/public';
+import { FormattedMessage } from '@kbn/i18n-react';
+import url from 'url';
+import type { ExportModalShareOpts } from '.';
+import { checkLicense } from '../..';
+
+export const reportingCsvShareProvider = ({
+  apiClient,
+  application,
+  license,
+  usesUiCapabilities,
+  theme,
+}: ExportModalShareOpts): ShareMenuProvider => {
+  const getShareMenuItems = ({
+    objectType,
+    objectId,
+    sharingData,
+    onClose,
+    intl,
+    toasts,
+  }: ShareContext) => {
+    if ('search' !== objectType) {
+      return [];
+    }
+
+    // only csv v2 supports esql (isTextBased) reports
+    // TODO: whole csv reporting should move to v2 https://github.com/elastic/kibana/issues/151190
+    const reportType = sharingData.isTextBased ? CSV_JOB_TYPE_V2 : CSV_JOB_TYPE;
+
+    const getSearchSource = sharingData.getSearchSource as ({
+      addGlobalTimeFilter,
+      absoluteTime,
+    }: {
+      addGlobalTimeFilter?: boolean;
+      absoluteTime?: boolean;
+    }) => SearchSourceFields;
+
+    const jobParams = {
+      title: sharingData.title as string,
+      objectType,
+    };
+
+    const getJobParams = (forShareUrl?: boolean) => {
+      if (reportType === CSV_JOB_TYPE_V2) {
+        // csv v2 uses locator params
+        return {
+          ...jobParams,
+          locatorParams: sharingData.locatorParams as [Record<string, unknown>],
+        };
+      }
+
+      // csv v1 uses search source and columns
+      return {
+        ...jobParams,
+        columns: sharingData.columns as string[] | undefined,
+        searchSource: getSearchSource({
+          addGlobalTimeFilter: true,
+          absoluteTime: !forShareUrl,
+        }),
+      };
+    };
+
+    const shareActions: ShareMenuItem[] = [];
+
+    const licenseCheck = checkLicense(license.check('reporting', 'basic'));
+    const licenseToolTipContent = licenseCheck.message;
+    const licenseHasCsvReporting = licenseCheck.showLinks;
+    const licenseDisabled = !licenseCheck.enableLinks;
+
+    let capabilityHasCsvReporting = false;
+    if (usesUiCapabilities) {
+      capabilityHasCsvReporting = application.capabilities.discover?.generateCsv === true;
+    } else {
+      capabilityHasCsvReporting = true; // deprecated
+    }
+
+    const generateReportingJobCSV = ({ intl: intlReport, toasts: toastsReport }: ShareContext) => {
```

Comment:
The services are already passed into `getShareMenuItems`, so we don't need to use the params to this function.
```suggestion
    const generateReportingJobCSV = () => {
```

Instead of `toastsReport`, you can just `toasts` and instead of `intlReport`, you can use `intl`